framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -859,59 +904,84 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
 
-                    if (sleep_time < kMaxSleepLimitNs)
-                    {
-                        // Next time, sleep for double what we initially slept for.  This way if we're just
-                        // spinning, we spin less and less each time.
-                        sleep_time *= 2;
-                    }
-                    // Clamp it to the max (should stay here from this point forward)
-                    if (sleep_time > kMaxSleepLimitNs)
-                    {
-                        sleep_time = kMaxSleepLimitNs;
-                    }
-                }
-                else
-                {
-                    GFXRECON_LOG_ERROR("xrPollEvent encountered an error of type 0x%x", replay_result);
-                    break;
-                }
+                // Next time, sleep for double what we initially slept for.  This way if we're just
+                // spinning, we spin less and less each time.
+                sleep_time *= 2;
+            }
+            else
+            {
+                GFXRECON_LOG_ERROR("xrPollEvent encountered an error of type 0x%x", replay_result);
+                break;
             }
         } while ((retry_count < kRetryLimit) && capture_event->type != out_eventData->type);
+
+        // We timed out and never got the event we expect, so handle it appropriately.
         if (capture_event->type != out_eventData->type)
         {
-            GFXRECON_LOG_ERROR("Event %s (0x%x %d) never occurred!",
-                               GetEventTypeString(capture_event->type),
-                               capture_event->type,
-                               capture_event->type);
-            replay_result = XR_ERROR_RUNTIME_FAILURE; // Runtime never gave us the event we were looking for
+            // If the event that is expected is not known to  us, and we didn't encounter it , then this is a
+            // bad situtation and needs to be reported.
+            if (IsHandledEventType(capture_event->type))
+            {
+                GFXRECON_LOG_ERROR("Event %s (0x%x %d) never occurred!",
+                                   GetEventTypeString(capture_event->type),
+                                   capture_event->type,
+                                   capture_event->type);
+
+                // Runtime never gave us the event we were looking for
+                replay_result = XR_ERROR_RUNTIME_FAILURE;
+            }
+            else if (options_.ignore_unhandled_unknown_capture_events)
+            {
+                // If this is an event we don't know anything about (it is unknown to GFXR), then
+                // add it to the list of events we check against in the future.
+                GFXRECON_LOG_WARNING(
+                    "Expected Capture Event %s (0x%x %d) did not occur, but since it is unknown, it will be skipped",
+                    GetEventTypeString(capture_event->type),
```

Comment:
Since this is on the else of "IsHandled..." isn't this guaranteed to be "Unknown Event"